### ISSUE TITLE
Videojs-ima throws error in console but still runs all breakpoints

### ISSUE THREAD

**Issue by anton-asnitsky** [googleads/videojs-ima#696]
Hi all,

I've faced strange problem.
When player started, ima plugin throws an error to console, but still runs all requested breakpoints(pre-, mid- and post-rolls).
The error was thrown:
6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:7408 VIDEOJS: ERROR: TypeError: Cannot read property 'requestAds' of null
    at SdkImpl.requestAds (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:82178)
    at Controller.requestAds (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:82828)
    at ImaPlugin.<anonymous> (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:83474)
    at Player.<anonymous> (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:84490)
    at HTMLDivElement.bound (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:9397)
    at HTMLDivElement.func (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:9223)
    at HTMLDivElement.data.dispatcher (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:9044)
    at trigger (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:9176)
    at Player.trigger$$1 [as trigger] (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:10015)
    at Player.<anonymous> (6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:10949)
logByType @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:7408
log.error @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:7529
data.dispatcher @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:9046
trigger @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:9176
trigger$$1 @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:10015
(anonymous) @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:10949
bound @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:9397
setTimeout (async)
setTimeout @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:11599
triggerReady @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:10930
handleTechReady_ @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:29635
bound @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:9397
ready @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:10912
loadTech_ @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:29454
MediaLoader @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:18759
addChild @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:10699
handleAdd @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:10821
initChildren @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:10868
Player @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:28888
videojs$1 @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:33153
window.WPlayer @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:84273
Walla.WallaPlayer @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:85321
Walla.WallaPlayer.init @ 6_8_12_17_19_23_24_25_26_27_29_30_31_33_34_35_40_43_48_51_52_54_57_67_69_73_74_76_79_82_88_90_108_112_114_136_179_180_183_184_215_216_217_218_219_220_221_222_223_224_225_226_227_228_229.js?b=1517130105:85351
(anonymous) @ common.js?b=1517130105:665
each @ common.js?b=1517130105:5
each @ common.js?b=1517130105:5
(anonymous) @ common.js?b=1517130105:655
j @ common.js?b=1517130105:5
fireWith @ common.js?b=1517130105:5
ready @ common.js?b=1517130105:5
J @ common.js?b=1517130105:5

What can it be?

URL to check:  http://celebs.dfp-ads-on-video.walla.news/item/3167020
Videojs version: 7.0.5
videojs-contrib-ads version: 6.4.1
videojs-ima version: 1.5.1

**Comment by anton-asnitsky** [comment]
![image](https://user-images.githubusercontent.com/8492824/43065460-9631eea2-8e6a-11e8-90ef-23a246bb9238.png)

**Comment by ypavlotsky** [comment]
Hi, I'm getting a 503 error when I go to your page. Are there any other details on how to reproduce the errors you're seeing?

**Comment by anton-asnitsky** [comment]
Can you tell me on which hours yo're checking this address? Our qa env. shuts down at night by schedule hours to save money :)

**Comment by anton-asnitsky** [comment]
Hi there' here the working url for checks: http://celebs.stage.walla.co.il/item/3167020
VErsions are updated:
"video.js": "7.2.0",
 "videojs-contrib-ads": "6.4.3",
"videojs-ima": "1.5.1"

**Comment by davidjamesherzog** [comment]
@anton-asnitsky - I had a similar type issue.  What I realized is that the plugin calls `requestAds` if the player is ready https://github.com/googleads/videojs-ima/blob/master/src/sdk-impl.js#L556-L559.  I had a race condition with API calls where we would call `requestAds` prior to the video.js player being ready.  I created a flag to say whether or not the player was ready and used that to determine whether or not to call requestAds.  If the player is not ready when you want to call `requestAds`, the plugin should call it for you when the player is ready.  Hope this helps you.
